File: src/minecraft/me/skrilled/utils/config/AbstractConfig.java
package me.skrilled.utils.config;

import me.skrilled.SenseHeader;

import java.nio.file.Path;
import java.nio.file.Paths;

public abstract class AbstractConfig {

    private final String name;
    private final Path path;

    public AbstractConfig() {
        this.name = this.getClass().getAnnotation(ConfigData.class).name();
        this.path = Paths.get(String.valueOf(SenseHeader.getSense.getDirectory()), "configs", this.getName() + ".sense");
    }

    public abstract void load();

    public abstract void save();

    public String getName() {
        return name;
    }

    public Path getPath() {
        return path;
    }
}

File: minecraft/me/kl0udy92/apart/config/AbstractConfig.kt
package me.kl0udy92.apart.config

import me.kl0udy92.apart.config.annotations.ConfigData
import me.kl0udy92.apart.config.interfaces.IConfig
import me.kl0udy92.apart.core.Main
import java.nio.file.Path
import java.nio.file.Paths

abstract class AbstractConfig : IConfig {

    val name: String
    val path: Path

    init {
        this.name = this.javaClass.getAnnotation(ConfigData::class.java).name
        this.path = Paths.get(Main.directory.toString(), "configs", "${this.name}.json")
    }

}